editor/src/components/canvas/canvas-utils.ts

```diff
@@ -2617,3 +2622,61 @@ export function getValidElementPathsFromElement(
     return []
   }
 }
+
+function createCanvasTransientStateFromProperties(
+  editor: EditorState,
+): TransientCanvasState | null {
+  if (editor.canvas.transientProperties == null) {
+    return null
+  } else {
+    const updatedEditor = Object.values(editor.canvas.transientProperties).reduce(
+      (working, currentProp) => {
+        return modifyUnderlyingTarget(
+          currentProp.elementPath,
+          Utils.forceNotNull('No open file found', getOpenUIJSFileKey(editor)),
+          working,
+          (element: JSXElement) => {
+            const valuesAtPath = Object.keys(currentProp.attributesToUpdate).map((key) => {
+              return {
+                path: PP.fromString(key),
+                value: currentProp.attributesToUpdate[key],
+              }
+            })
+            let updatedAttributes = setJSXValuesAtPaths(element.props, valuesAtPath)
+            return foldEither(
+              (_) => element,
+              (updatedProps) => {
+                return {
+                  ...element,
+                  props: updatedProps,
+                }
+              },
+              updatedAttributes,
+            )
+          },
+        )
+      },
+      editor,
+    )
+
+    let transientFilesState: TransientFilesState = {}
+    fastForEach(Object.values(editor.canvas.transientProperties) ?? [], (prop) => {
+      forUnderlyingTargetFromEditorState(
+        prop.elementPath,
+        updatedEditor,
+        (success, underlyingElement, underlyingTarget, underlyingFilePath) => {
+          transientFilesState[underlyingFilePath] = {
```

Comment:
if I have two `editor.canvas.transientProperties` entries pointing to the same file, will they overwrite each other and the last one win out?

Reply:
nope, the transientproperties are applied to the jsx elements in a reduce in a few lines above this based on elementpath

Reply:
I should add some tests

Reply:
or at least make yourself a TODO for tests :) but maybe not for today, as we are still figuring this feature out